src/instructlab/config.py

```diff
@@ -216,3 +223,50 @@ def get_model_family(forced, model_path):
     guess = MODEL_FAMILY_MAPPINGS.get(guess, guess)
 
     return guess if guess in MODEL_FAMILIES else DEFAULT_MODEL_FAMILY
+
+
+class Lab:
+    """Lab object holds high-level information about ilab CLI"""
+
+    def __init__(self, config_obj: Config):
+        self.config = config_obj
+
+        # Set up logging for the Lab class
+        self.logger = logging.getLogger(__name__)
+
+        # Create a formatter
+        formatter = log.CustomFormatter(log.FORMAT)
+
+        # Create a handler and set the formatter
+        handler = logging.StreamHandler()
+        handler.setFormatter(formatter)
+
+        logging.basicConfig(
+            format=log.FORMAT,
+            handlers=[handler],
+        )
+
+        self.logger.setLevel(self.config.general.log_level.upper())
+
+
+def init_config(ctx, config_file):
+    if (
+        ctx.invoked_subcommand not in {"config", "init", "sysinfo"}
+        and "--help" not in sys.argv[1:]
+    ):
+        if config_file == "DEFAULT":
+            config_obj = get_default_config()
+        elif not os.path.isfile(config_file):
+            config_obj = None
+            ctx.fail(
+                f"`{config_file}` does not exists, please run `ilab init` "
```

Comment:
Shouldn't this say please run `ilab config init`?

Reply:
... and should say "does not exist" instead of "exists"

but I don't think we need to block on this. It's pre-existing code and `ilab init` still works

Reply:
https://github.com/instructlab/instructlab/issues/1331